### ISSUE TITLE
VS integration not working?

### ISSUE THREAD

**Issue by adam-c-anderson** [deviousasti/fsharp-linting-for-vs#4]
Version: 0.2
Visual Studio version: 16.6.0

We've installed FSharpLint as a dotnet tool and configured a git pre-commit hook to lint files there. I also installed the F# Lint VS extension for earlier detection and fixes. Today I had a successful build with no linting errors in VS, but the pre-commit hook rejected the commit because the linter there caught a violation. 
![image](https://user-images.githubusercontent.com/26259680/96143514-27959400-0ed1-11eb-96c0-21ce6b64e386.png)

Additional things I noticed while trying to diagnose and gather info for this report:

The option page for the extension is completely blank
![image](https://user-images.githubusercontent.com/26259680/96142563-40ea1080-0ed0-11eb-8584-0d8946e1d2d1.png)

The version appears a 0.2 in the extension manager but as 0.1 in Help | About Visual Studio
![image](https://user-images.githubusercontent.com/26259680/96142822-80186180-0ed0-11eb-883b-e3f8ea16719a.png)

![image](https://user-images.githubusercontent.com/26259680/96142877-8e667d80-0ed0-11eb-9a37-78558e75d1f7.png)

**Comment by deviousasti** [comment]
Hello Adam,
Thank you for the detailed report.

>  Today I had a successful build with no linting errors in VS

Linting the entire solution on build is extremely expensive. 
Generally, this would be done as either a release configuration [MSBuild task](https://fsprojects.github.io/FSharpLint/how-tos/msbuild-task.html) or a FAKE target. 

Linting for VS behaves similarly to linting for VS Code, and other linters like ts-lint, in that it only lints currently open files and lint warnings are expected to disappear as soon a file is closed. 

Would you like to have a menu option which lints the entire project?
The options page is currently a placeholder - I'm presently taking feedback on what should be configurable for F# lint.

P.S.
The warning in the screenshot seems like a formatting warning rather than a lint warning (I could be mistaken).
Could you tell me the lint rule (like FL00xxx) it didn't pick up, so I could reproduce it?

Regards.

**Comment by adam-c-anderson** [comment]
Well, to be more precise, I simply don't see any linting feedback in VS at all. 😄 I opened the file that contains the linting error but the violation isn't highlighted. We have this rule configured in our `fsharplint.json` which is in the same directory as the sln:

```json
    "typedItemSpacing": {
        "enabled": true,
        "config": {
            "typedItemStyle": "SpaceAfter"
        }
    },
```

As shown in the console screenshot above, this is reported by the linter when run as a dotnet tool, but it does not appear in the errors list or with a green underbar in the editor when the file is open.

I can't see a reference to a rule number in the format `FL00xxx`, but this appears to be the corresponding rule: https://fsprojects.github.io/FSharpLint/reference/FSharpLint.Core/fsharplint-rules-typeditemspacing.html

Thanks!

**Comment by deviousasti** [comment]
I'll release an update today. All shall be fixed. 😄

**Comment by deviousasti** [comment]
I've released an update which should support formatting and type-checked linting rules.
And I've added some much needed options to the options page. 

Could you update and see if the issue's been resolved?

**Comment by adam-c-anderson** [comment]
Confirmed; I installed the update and immediately was able to see the linting feedback both in-editor and in the errors list. Thanks!

**Comment by deviousasti** [comment]
If there's nothing else, shall we close this issue?

**Comment by adam-c-anderson** [comment]
👍
